web/src/auth/SignupPage.js

```diff
@@ -812,9 +816,12 @@ class SignupPage extends React.Component {
 
   render() {
     const application = this.getApplicationObj();
-    if (application === undefined || application === null) {
+    if (application === undefined) {
       return null;
     }
+    if (application === null) {
```

Comment:
Why not show error in EntryPage.js ? It's the entrance for pages like signup, login, forget-password, etc. So only one hook is needed